src/rdkafka_metadata.c

```diff
@@ -1971,3 +1971,179 @@ rd_kafka_metadata_new_topic_with_partition_replicas_mock(int replication_factor,
         return rd_kafka_metadata_new_topic_mock(
             topics, topic_cnt, replication_factor, num_brokers);
 }
+
+/**
+ * @brief Handle update of metadata received in the produce or fetch tags.
+ *
+ * @param rk Client instance.
+ * @param rko Metadata update operation.
+ *
+ * @locality main thread
+ * @locks none
+ *
+ * @return always RD_KAFKA_OP_RES_HANDLED
+ */
+rd_kafka_op_res_t
+rd_kafka_metadata_update_op(rd_kafka_t *rk, rd_kafka_metadata_internal_t *mdi) {
+        int i, j;
+        rd_kafka_metadata_t *md = &mdi->metadata;
+        rd_kafka_broker_t *rkb;
+        rd_bool_t cache_updated = rd_false;
+        rd_kafka_metadata_partition_t *mdp;
+        rd_kafka_metadata_partition_internal_t *mdpi;
+        rd_kafka_secproto_t rkb_proto = rk->rk_conf.security_protocol;
+
+
+        for (i = 0; i < md->broker_cnt; i++) {
+                rd_kafka_broker_update(rk, rkb_proto, &md->brokers[i], NULL);
+        }
+
+        struct rd_kafka_metadata_cache_entry *rkmce;
+        int32_t partition_cache_changes = 0, part, current_leader_epoch;
+        rd_bool_t by_id                 = rd_false;
+        rd_kafka_Uuid_t topic_id;
+        char *topic;
```

Comment:
These variables need to be at the beginning of the for loop below, where they're used.

Reply:
Updated